webapp/src/shortcut_formatting.ts

```diff
@@ -0,0 +1,116 @@
+import { ShortcutRegistry } from 'blockly';
+import { keyNames } from './keynames';
+
+const isMacPlatform = pxt.BrowserUtils.isMac();
+
+/**
+ * Default keyboard navigation shortcut names.
+ * Based from blockly-keyboard-experiment constants.ts.
+ */
+export enum SHORTCUT_NAMES {
+  UP = 'up',
+  DOWN = 'down',
+  RIGHT = 'right',
+  LEFT = 'left',
+  INSERT = 'insert',
+  EDIT_OR_CONFIRM = 'edit_or_confirm',
+  DISCONNECT = 'disconnect',
+  TOOLBOX = 'toolbox',
+  EXIT = 'exit',
+  MENU = 'menu',
+  COPY = 'keyboard_nav_copy',
+  CUT = 'keyboard_nav_cut',
+  PASTE = 'keyboard_nav_paste',
+  DELETE = 'keyboard_nav_delete',
+  MOVE_WS_CURSOR_UP = 'workspace_up',
+  MOVE_WS_CURSOR_DOWN = 'workspace_down',
+  MOVE_WS_CURSOR_LEFT = 'workspace_left',
+  MOVE_WS_CURSOR_RIGHT = 'workspace_right',
+  CREATE_WS_CURSOR = 'to_workspace',
+  LIST_SHORTCUTS = 'list_shortcuts',
+  CLEAN_UP = 'clean_up_workspace',
+  UNDO = 'undo',
+  REDO = 'redo'
+}
+
+export function getActionShortcut(action: string): string[] {
+  return getActionShortcutsAsKeys(action, shortModifierNames)[0]
+          .map(s => displayArrowKeys[s] ?? s);
+}
+
+const displayArrowKeys: Record<string, string> = {
+  "Up": "↑",
+  "Down": "↓",
+  "Left": "←",
+  "Right": "→"
+}
+
+// Copied from blockly-keyboard-experiment.
+const shortModifierNames: Record<string, string> = {
+  'Control': 'Ctrl',
+  'Meta': '⌘',
+  'Alt': isMacPlatform ? '⌥' : 'Alt',
+};
+
+/**
+ * Find the relevant shortcuts for the given action for the current platform.
+ * Keys are returned in a user facing format.
+ *
+ * This could be considerably simpler if we only bound shortcuts relevant to the
+ * current platform or tagged them with a platform.
+ * 
+ * Copied from blockly-keyboard-experiment.
+ *
+ * @param action The action name, e.g. "cut".
+ * @param modifierNames The names to use for the Meta/Control/Alt modifiers.
+ * @returns The formatted shortcuts.
+ */
+function getActionShortcutsAsKeys(
```

Comment:
This is all a bit gross but matches the keyboard nav plugin's code for its simpler keyboard shortcut dialog.

As per the comment above, it would all be much nicer if Blockly only bound shortcuts relevant to the current platform or at least tagged them with a platform.